### PR TITLE
Included net-5.0 specific updates for Modern Web Applications with ASP.NET Core - eBook

### PR DESCRIPTION
**Pull request by sughosneo**
This PR updates following pages :

- `Characteristics of modern web applications`
- `Choose Between Traditional Web Apps and Single Page Apps (SPAs)`
- `Architectural principles`
- `Azure hosting recommendations for ASP.NET Core web apps`
- `Common web application architectures`
- `Common client-side web technologies`
- `Developing ASP.NET Core MVC apps`
- `Development process for Azure`
- `Working with Data in ASP.NET Core Apps`
- `Test ASP.NET Core MVC apps`

### CODE REVIEW

**Comment by sughosneo** [comment]
Please note, Index page is not yet modified. It will get updated once the other changes are approved.

File: docs/architecture/modern-web-apps-azure/common-client-side-web-technologies.md [hunk 1]
```diff
@@ -154,23 +154,33 @@ This is enough to render "Hello Vue!" on the page. Note, however, that Vue isn't
 
 Of course, this only scratches the surface of what Vue is capable of. It's gained a great deal of popularity in the last several years and has a large community. There's a [huge and growing list of supporting components and libraries](https://github.com/vuejs/awesome-vue#redux) that work with Vue to extend it as well. If you're looking to add client-side behavior to your web application or considering building a full SPA, Vue is worth investigating.
 
+### Blazor WebAssembly
+
+Unlike other JavaScript frameworks, `Blazor WebAssembly` is a single-page app (SPA) framework for building interactive client-side web apps with .NET. Blazor WebAssembly uses open web standards without plugins or recompiling code into other languages. Blazor WebAssembly works in all modern web browsers, including mobile browsers.
+
+Running .NET code inside web browsers is made possible by WebAssembly (abbreviated `wasm`). WebAssembly is a compact bytecode format optimized for fast download and maximum execution speed. WebAssembly is an open web standard and supported in web browsers without plugins.
```
Comment (ardalis) [inline docs/architecture/modern-web-apps-azure/common-client-side-web-technologies.md:161]:
```suggestion
Running .NET code inside web browsers is made possible by WebAssembly (abbreviated `wasm`). WebAssembly is a compact bytecode format optimized for fast download and maximum execution speed. WebAssembly is an open web standard and is supported in web browsers without plugins.
```

File: docs/architecture/modern-web-apps-azure/common-client-side-web-technologies.md [hunk 2]
```diff
@@ -154,23 +154,33 @@ This is enough to render "Hello Vue!" on the page. Note, however, that Vue isn't
...
 
-When considering which JavaScript framework will work best to support your SPA, keep in mind the following considerations:
+When considering which option will work best to support your SPA, keep in mind the following considerations:
 
 - Is your team familiar with the framework and its dependencies (including TypeScript in some cases)?
 
 - How opinionated is the framework, and do you agree with its default way of doing things?
 
-- Does it (or a companion library) include all of the features your app requires?
+- Does it (or a companion library) includes all of the features your app requires?
```
Comment (ardalis) [inline docs/architecture/modern-web-apps-azure/common-client-side-web-technologies.md:175]: I wouldn't make this change. "Does it includes all of the features" doesn't make sense to me.

File: docs/architecture/modern-web-apps-azure/modern-web-applications-characteristics.md [hunk 3]
```diff
@@ -57,11 +57,11 @@ Many web applications involve a combination of traditional web application behav
...
+ASP.NET Core applications can be written using simple text editors and command-line interfaces, or full-featured development environments like Visual Studio. Monolithic applications are typically deployed to a single endpoint. Deployments can easily be automated to occur as part of continuous integration (CI) and continuous delivery (CD) pipeline. In addition to traditional CI/CD tools, Microsoft Azure has integrated support for git repositories and can automatically deploy updates as they are made to a specified git branch or tag. Azure DevOps provides a full-featured CI/CD build and deployment pipeline, and GitHub Actions provide another option for projects hosted there.
```
Comment (ardalis) [inline docs/architecture/modern-web-apps-azure/modern-web-applications-characteristics.md:60]: This change only works if you get rid of "pipeline" at the end. Deployments can be part of CI or part of CD but they need to be part of *a* CI or CD *pipeline*.

File: docs/architecture/modern-web-apps-azure/modern-web-applications-characteristics.md [hunk 4]
```diff
@@ -57,11 +57,11 @@ Many web applications involve a combination of traditional web application behav
...
 
 ## Traditional ASP.NET and Web Forms
 
-In addition to ASP.NET Core, traditional ASP.NET 4.x continues to be a robust and reliable platform for building web applications. ASP.NET supports MVC and Web API development models, as well as Web Forms, which is well suited to rich page-based application development and features a rich third-party component ecosystem. Microsoft Azure has great longstanding support for ASP.NET 4.x applications, and many developers are familiar with this platform.
+In addition to ASP.NET Core, traditional ASP.NET 4.x continues to be a robust and reliable platform for building web applications. ASP.NET supports MVC and Web API development models, as well as Web Forms, which are well suited to rich page-based application development and features a rich third-party component ecosystem. Microsoft Azure has great longstanding support for ASP.NET 4.x applications, and many developers are familiar with this platform.
```
Comment (ardalis) [inline docs/architecture/modern-web-apps-azure/modern-web-applications-characteristics.md:64]: If we're going to say "Web Apps for Containers" is singular, then the product "Web Forms" is also singular.

**Review by ardalis — CHANGES_REQUESTED** [review]
A few minor changes requested, thanks!

File: docs/architecture/modern-web-apps-azure/common-client-side-web-technologies.md (hunk 2 above)
Comment (IEvangelist) [reply docs/architecture/modern-web-apps-azure/common-client-side-web-technologies.md:175]: I agree with @ardalis here, I'd leave it as "include"
```suggestion
- Does it (or a companion library) include all of the features your app requires?
```

File: docs/architecture/modern-web-apps-azure/modern-web-applications-characteristics.md (hunk 4 above)
Comment (sughosneo) [reply docs/architecture/modern-web-apps-azure/modern-web-applications-characteristics.md]: Yes, that makes sense. Reverted back the change.

File: docs/architecture/modern-web-apps-azure/modern-web-applications-characteristics.md (hunk 3 above)
Comment (sughosneo) [reply docs/architecture/modern-web-apps-azure/modern-web-applications-characteristics.md]: I see. Made the changes again.

**Comment by sughosneo** [comment]
> A few minor changes requested, thanks!

Thanks @ardalis  for your feedback. 
I have incorporated all the suggestions and also updated the `index` page for further review.

**Review by ardalis — APPROVED** [review]

**Review by nishanil — APPROVED** [review]

**Pull request merged**